pkg/tsdb/query_context.go

```diff
@@ -6,9 +6,9 @@ type QueryContext struct {
 	TimeRange   *TimeRange
 	Queries     QuerySlice
 	Results     map[string]*QueryResult
-	ResultsChan chan *BatchResult
-	Lock        sync.RWMutex
-	BatchWaits  sync.WaitGroup
+	ResultsChan chan *BatchResult `json:"-"`
+	Lock        sync.RWMutex      `json:"-"`
+	BatchWaits  sync.WaitGroup    `json:"-"`
```

Comment:
Ah, I was initially marshaling the hole thing; but I stopped doing that a while ago; good catch.